packages/truffle-box/box.js

```diff
@@ -111,21 +110,22 @@ const Box = {
       tmp.setGracefulCleanup();
     }
 
-    tmp.dir({ unsafeCleanup }, function(err, dir) {
-      if (err) return callback(err);
-
-      self
-        .unbox(
-          "https://github.com/trufflesuite/truffle-init-" + name,
-          dir,
-          options
-        )
-        .then(function() {
-          var config = Config.load(path.join(dir, "truffle-config.js"), {});
-          callback(null, config);
-        })
-        .catch(callback);
-    });
+    try {
+      const tmpDir = tmp.dirSync({ unsafeCleanup });
+      Box.unbox(
+        `https://github.com/trufflesuite/truffle-init-${name}`,
+        tmpDir.name,
+        options
+      ).then(() => {
+        const config = Config.load(
+          path.join(tmpDir.name, "truffle-config.js"),
+          {}
+        );
+        return callback(null, config);
+      });
```

Comment:
I think you will need to keep a catch here that just throws the error, otherwise node will whine about an unhandled Promise rejection in the case of an error. So throw the error in the catch and it will be handled in the catch of the try/catch.